### PR TITLE
Use standalone kubectl for gke, remove special case for vagrant

### PR DESCRIPTION
**Pull request by j3ffml**
also fix hack/local-up-cluster.sh kubeconfig instructions

### CODE REVIEW

File: cluster/kube-env.sh [hunk 1]
```diff
@@ -20,7 +20,7 @@
 #
 # The valid values: 'gce', 'gke', 'aws', 'azure', 'vagrant', 'vsphere', 'libvirt-coreos', 'juju'
 
-KUBERNETES_PROVIDER=${KUBERNETES_PROVIDER:-gce}
+KUBERNETES_PROVIDER=${KUBERNETES_PROVIDER:-gke}
```
Comment (roberthbailey) [inline cluster/kube-env.sh]: Did you mean to change the default to gke?

**Comment by roberthbailey** [comment]
Other than the question about the default, LGTM.

File: cluster/kube-env.sh (hunk 1 above)
Comment (jlowdermilk) [inline cluster/kube-env.sh]: Nope, accidental commit.

**Comment by roberthbailey** [comment]
LGTM. Will merge on green.

**Comment by derekwaynecarr** [comment]
I must be missing something.  Why is the vagrant stanza no longer needed?

**Comment by jlowdermilk** [comment]
The vagrant stanza is no longer needed because `kube-up.sh` for vagrant generates kubeconfig using the same method shared by all providers, so no special handling is needed. See #6691

**Pull request merged**